src/interpreter/interpreter.rs

```diff
@@ -490,6 +532,15 @@ impl<R: RealNumberInternalTrait, E: IEnvironment<R>> Interpreter<R, E> {
         }
         Ok(Library::new(name, definitions))
     }
+    pub fn eval_library_from_char_stream(
+        &mut self,
+        char_stream: impl Iterator<Item = char>,
+    ) -> Result<Library<R, E>> {
+        let lexer = Lexer::from_char_stream(char_stream);
+        let parser = Parser::from_lexer(lexer);
+        let statements = parser.into_iter().collect::<Result<Vec<_>>>()?;
```

Comment:
Collecting to a vec is unnecessary memory comsuming, should replace with try_fold